hawk-java/src/main/java/so/hawk/java/catcher/Hawk.java

```diff
@@ -0,0 +1,151 @@
+ */
+public class Hawk {
+    private static volatile Hawk instance;
+    private final CustomUncaughtExceptionHandler exceptionHandler;
+    private String token;
+    private String secret;
+    private String integrationId;
+    private static String endpointBase;
+
+    /**
+     * Returns the singleton instance of HawkCatcher.
+     *
+     * @return the singleton instance
+     */
+
+    /**
+     * Initializes a new HawkCatcher instance.
+     */
+    private Hawk(String token){
+        this.token = token;
+        decodeToken(token);
+        endpointBase = String.format("https://%s.k1.hawk.so", integrationId);
+        this.exceptionHandler = new CustomUncaughtExceptionHandler();
+    }
+
+    // Initialize Hawk with a token (must be called first)
+    public static synchronized void init(String token) {
+        if (instance == null) {
+            instance = new Hawk(token);
+        }
+        getInstance().exceptionHandler.enable();
+    }
+
+    // Get the Singleton instance
+    private static Hawk getInstance() {
+        if (instance == null) {
+            throw new IllegalStateException("Hawk is not initialized. Please call Hawk.init(token) before using.");
+        }
+        return instance;
+    }
+    /**
+     *  Sets the custom handler as the default uncaught exception handler.
+     */
+
+    private void decodeToken(String token){
+        try {
+            String decodedJson = new String(Base64.getDecoder().decode(token));
+            JSONObject json = new JSONObject(decodedJson);
+            this.integrationId = json.getString("integrationId");
+            this.secret = json.getString("secret");
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Invalid token: Unable to decode Base64 JSON.");
+        }
+    }
+
+    public static void logError(Exception e) {
+        Hawk hawk = getInstance();
+        String payload = new JSONObject()
+            .put("token", hawk.token)
+            .put("catcherType", "errors/java")
+            .put("payload", new JSONObject()
+                .put("title", e.toString())
+                .put("backtrace", hawk.getStackTrace(e))
+            )
+            .toString();
+
+        System.out.println(e.toString());
+
+
+        hawk.sendPostRequest(payload);
+        System.out.println("Send log");
+    }
+
+    public static void send(String message) {
+        Hawk hawk = getInstance();
+        String payload = new JSONObject()
+            .put("token", hawk.token)
+            .put("catcherType", "errors/java")
+            .put("payload", new JSONObject()
+                .put("title", message)
+//                .put("backtrace", new JSONArray())) // Empty backtrace for custom messages
+            )
+            .toString();
+        hawk.sendPostRequest(payload);
+        System.out.println("Send succsess");
+    }
+
+    // Helper method to convert stack trace to a JSON array
+    private JSONArray getStackTrace(Exception e) {
```

Comment:
it should satisfy our format, not just array of strings. I'd suggest to remove it for now and implement in a separate PR